finetune/target_models/masked_language_model.py

```diff
@@ -38,17 +38,29 @@ def feed_shape_type_def(self):
             (shapes,),
         )
 
-    def text_to_tokens_mask(self, X, Y=None, context=None):
+    def text_to_tokens_mask(self, X, Y=None, context=None, forced_mask=None):
         out_gen = self._text_to_ids(X, pad_token=self.config.pad_token)
 
         for out in out_gen:
             seq_len = out.token_ids.shape[0]
-            mlm_mask = np.random.rand(seq_len) < self.config.mask_proba
-            mask_type = np.random.choice(
-                ["mask", "random", "unchanged"],
-                size=seq_len,
-                p=[0.8, 0.1, 0.1]
-            )
+            if forced_mask:
+                try:
+                    # we need to align the indico-style mask with the tokenized text
+                    # in the same way as we do for context
+                    mlm_mask = tokenize_masking(forced_mask, out).reshape(-1)
```

Comment:
I feel pretty strongly that we should be able to reuse the same core routine tokenize_context does here?